examples/test/reducer.py

```diff
@@ -1,16 +1,40 @@
 import logging
 
+from readerwriterlock import rwlock
+
 from dranspose.event import ResultData
+from dranspose.protocol import ReducerState
 
 
 class TestReducer:
-    def __init__(self, state=None, **kwargs):
-        self.publish = {"results": {}, "parameters": {}}
+    def __init__(self, state: ReducerState | None = None, **kwargs: dict) -> None:
+        self.publish: dict[str, dict] = {"results": {}, "parameters": {}}
 
-    def process_result(self, result: ResultData, parameters=None):
+    def process_result(
+        self, result: ResultData, parameters: dict | None = None
+    ) -> None:
         logging.info("parameters are %s", parameters)
         self.publish["results"][result.event_number] = result.payload
         self.publish["parameters"][result.event_number] = parameters
 
-    def finish(self, parameters=None):
+    def finish(self, parameters: dict | None = None) -> None:
+        print("finished dummy reducer work")
+
+
+class LockReducer:
```

Comment:
This class is never used. I like your way of putting the payload into the test way better than having the tests depend on the examples.